File: BAEKJOON/2231.swift
import Foundation

let n = Int(readLine()!)!
var sum = 0
var value = 0


for i in 1..<n {
    sum = i
    for j in String(i) {
        sum +=Int(String(j))!
    }
    if sum == n {
        value = i
        break
    }
}

print(value == 0 ? "0" : "\(value)")

File: BAEKJOON/2231.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int sum = 0;
        int value = 0;

        for(int i = 1; i < n; i++) {
            int number = i;
            sum = i;
            while(number != 0) {
                sum += number % 10;
                number /= 10;
            }
            if(sum == n) {
                value = i;
                break;
            }
        }
        System.out.println(value);

    }
}